modules/template/services/ImportInstanceService.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\custom_pages\modules\template\services;
+
+use humhub\modules\custom_pages\modules\template\elements\BaseElementContent;
+use humhub\modules\custom_pages\modules\template\elements\ContainerElement;
+use humhub\modules\custom_pages\modules\template\elements\ContainerItem;
+use humhub\modules\custom_pages\modules\template\models\Template;
+use humhub\modules\custom_pages\modules\template\models\TemplateElement;
+use humhub\modules\custom_pages\modules\template\models\TemplateInstance;
+use Yii;
+
+/**
+ * Service to import Template Instance (Custom Page or Container Item)
+ */
+class ImportInstanceService extends BaseImportService
+{
+    private TemplateInstance $instance;
+    private ?TemplateElement $element = null;
+
+    public function __construct(TemplateInstance $instance, ?TemplateElement $element = null)
+    {
+        $this->instance = $instance;
+        $this->element = $element;
```

Comment:
Maybe some check here, that the element exists in the instance template?

Reply:
@luke- Done https://github.com/humhub/custom-pages/pull/407/commits/767203e94a9a481917c49da8527f42c58268c909#diff-58dc4003cf2898a4eb1983b594073860216d3d8c0267cbd644165e1b97f1064aR30-R33